query/src/sql/planner/plans/scalar.rs

```diff
@@ -410,7 +411,13 @@ pub struct ComparisonExpr {
 
 impl ScalarExpr for ComparisonExpr {
     fn data_type(&self) -> DataTypeImpl {
-        BooleanType::new_impl()
+        let mut data_type = BooleanType::new_impl();
+        if self.left.data_type() == DataTypeImpl::Null(NullType {})
```

Comment:
But we should also have `nullable(BooleanType)`

Reply:
For expr, do we need to consider `nullable`? Could you please give a specific case?

Reply:
You can add a `return_type` field for `ComparisonExpr` to store the resolved return type as `FunctionCall`.